include/openssl/ec.h

```diff
@@ -1457,7 +1457,8 @@ void EC_KEY_METHOD_get_verify(const EC_KEY_METHOD *meth,
 # define EVP_PKEY_CTRL_GET1_ID_LEN                       (EVP_PKEY_ALG_CTRL + 13)
 /* KDF types */
 # define EVP_PKEY_ECDH_KDF_NONE                          1
-# define EVP_PKEY_ECDH_KDF_X9_62                         2
+# define EVP_PKEY_ECDH_KDF_X9_63                         2
+# define EVP_PKEY_ECDH_KDF_X9_62   EVP_PKEY_ECDH_KDF_X9_63
```

Comment:
I'd put a comment here explaining that the original name was an error and is kept here for backwards compatibility reasons. I'd also suggest that if we're going to make this change then we should also update all internal uses of `EVP_PKEY_ECDH_KDF_X9_62` to use `EVP_PKEY_ECDH_KDF_X9_63` instead. For bonus points we should create a new `ECDH_KDF_X9_63()` function that replaces the `ECDH_KDF_X9_62()` function. We'll still need `ECDH_KDF_X9_62()` but it should just call `ECDH_KDF_X9_63()` (again with a comment explaining that the original name was an error). `ECDH_KDF_X9_62()` is an exported public function. I would make `ECDH_KDF_X9_63()` an internal function only.

Reply:
What is the reasoning for making the macro identifier public but keeping the function internal? Since everything declared in _ec.h_ is considered public, which header should this new function be declared in?
The original one is used in ec_pmeth and sm2_crypt, both do not seem to share any internal header files.

Also, should internal functions appear in the _libcrypto.num_ file or is it only for exported symbols?

Reply:
> What is the reasoning for making the macro identifier public but keeping the function internal?

We don't want to break the API. So anything that was public before must continue to remain public. The use of `EVP_PKEY_ECDH_KDF_X9_63` still seems like a reasonable part of the EVP API. On the other hand the function `ECDH_KDF_X9_62()` completely sidesteps EVP and is part of the low-level ECDH API. The strategic direction for the low-level APIs is to not export them. All new algorithms only get an EVP interface (e.g. the only way to invoke X448 is via the EVP API - you can't use the low-level primitives directly). Therefore my thinking is that `ECDH_KDF_X9_62()` will probably get deprecated at some point - so why create yet another low level function that we don't really want people to use as well? I've not got a strong view on this - so if you feel we should export it then I'm not going to complain too loudly.

> Since everything declared in ec.h is considered public, which header should this new function be declared in?

Probably crypto/include/internal/ec_int.h

> Also, should internal functions appear in the libcrypto.num file or is it only for exported symbols?

It's only for exported symbols. But you shouldn't need to edit it directly. Just invoke "make update" and it should get updated automatically.

Reply:
Yes we should start the name with lowercase.

If we add deprecation macros then I don't think we should backport that bit to 1.1.1. Otherwise I think everything else can be backported.